lib/highline/question/answer_converter.rb

```diff
@@ -0,0 +1,84 @@
+# coding: utf-8
+
+require 'forwardable'
+
+class HighLine
+  class Question
+    class AnswerConverter
+      extend Forwardable
+
+      def_delegators :@question,
+                     :answer, :answer=, :check_range,
+                     :directory, :answer_type, :choices_complete
+
+      def initialize(question)
+        @question = question
+      end
+
+      def convert
+        return unless answer_type
+
+        self.answer = convert_by_answer_type
+        check_range
+        answer
+      end
+
+      private
+
+      def convert_by_answer_type
+        if answer_type.respond_to? :parse
+          answer_type.parse(answer)
+        elsif answer_type.is_a? Class
+          send(answer_type.name)
+        else
+          send(answer_type.class.name)
+        end
+      end
+
+      def String
+        HighLine::String(answer)
+      end
+
+      # That's a weird name for a method!
+      # But it's working ;-)
+      define_method "HighLine::String" do
+        HighLine::String(answer)
+      end
+
+      def Integer
+        Kernel.send(:Integer, answer)
```

Comment:
```Integer(answer)``` is not working, I think it's because it tries to make a recursive call to the same Integer method.

But, inspired on your tip, I thoght about ```Kernel.Integer(answer)``` that is working great. It's simpler than using ```Kernel.send```

The ```send``` here is a heritage of the previous code.

As you can see, sometimes I just try to "extract" code with related logic into a proper class or method without trying to change it a lot. After that, with a contained piece of code I can think more clearly about how to really refactor it.

Your tips are great!